codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -587,7 +591,7 @@ private State readHeaders(ByteBuf buffer) {
 
         State nextState;
 
-        if (isContentAlwaysEmpty(message)) {
+        if (isContentAlwaysEmpty(message) || HttpUtil.hasEmptyContent(message)) {
```

Comment:
@jza70 I think this is not really correct as if a message has `Content-Length: 0` set it be preserved. Don't you think ?